### PR TITLE
[GEOS-9785] Invalid argument type=null when trying to use gs:Download WPS identifier

### PR DESCRIPTION
**Pull request by sergitopo**
[GEOS-9785](https://osgeo-org.atlassian.net/browse/GEOS-9785)

WPS download fails to estimate either the response result will exceed the size limit configured or not. Then no download process is executed.

This bug occurs when there is a Coverage saved in catalog which has missing info regarding the pixel dimension type in one or more bands. Unluckily this happens for the default geoserver coverages included in demo data.

# Checklist

- [x] I have read the [contribution guidelines](https://github.com/geoserver/geoserver/blob/main/CONTRIBUTING.md).
- [x] I have sent a [Contribution Licence Agreement](https://docs.geoserver.org/latest/en/developer/policies/committing.html) (not required for small changes, e.g., fixing typos in documentation).
- [x] First PR targets the `main` branch (backports managed later; ignore for branch specific issues).
- [ ] All the build checks are green ([see automated QA checks](https://docs.geoserver.org/latest/en/developer/qa-guide/index.html)).

For core and extension modules:

- [x] New unit tests have been added covering the changes.
- [ ] [Documentation](https://github.com/geoserver/geoserver/tree/main/doc/en/user/source) has been updated (if change is visible to end users).
- [ ] The [REST API docs](https://github.com/geoserver/geoserver/tree/main/doc/en/api/1.0.0) have been updated (when changing configuration objects or the REST controllers).
- [x] There is an issue in the [GeoServer Jira](https://osgeo-org.atlassian.net/browse/GEOS/summary) (except for changes that do not affect administrators or end users in any way).
- [x] Commit message(s) must be in the form ``[GEOS-XYZWV] Title of the Jira ticket``.
- [x] Bug fixes and small new features are presented as a single commit.
- [x] Each commit has a single objective (if there are multiple commits, each has a separate JIRA ticket describing its goal).

### CODE REVIEW

File: src/extension/wps-download/src/main/java/org/geoserver/wps/gs/download/RasterEstimator.java [hunk 1]
```diff
@@ -200,6 +200,14 @@ public boolean execute(
 
         // Use sample info type for each output band to estimate size
         List<CoverageDimensionInfo> coverageDimensionInfoList = coverageInfo.getDimensions();
+
+
+        if (coverageDimensionInfoList.stream().anyMatch(cdi -> cdi.getDimensionType() == null)) {
+            if (LOGGER.isLoggable(Level.FINE)) {
```
Comment (aaime) [inline src/extension/wps-download/src/main/java/org/geoserver/wps/gs/download/RasterEstimator.java:206]: The guard is not necessary when the message is a static string (no construction of new objects needed).

File: src/extension/wps-download/src/main/java/org/geoserver/wps/gs/download/RasterEstimator.java [hunk 2]
```diff
@@ -250,4 +258,24 @@ public boolean execute(
...
+            GridCoverage2DReader reader =
+                    (GridCoverage2DReader)
+                            catalog.getResourcePool()
+                                    .getGridCoverageReader(
+                                            ci, nativeName, GeoTools.getDefaultHints());
+            rebuilt = cb.buildCoverage(reader, nativeName, null);
+        } else {
+            rebuilt = cb.buildCoverage(nativeName);
+        }
+        ci.getDimensions().clear();
```
Comment (aaime) [inline src/extension/wps-download/src/main/java/org/geoserver/wps/gs/download/RasterEstimator.java:278]: This update is not really saving any new information in the catalog, and it's not intended to either (having config that gets modified as a result of a WPS process meant to download stuff would be unexpected).

Maybe it's best to just return rebuilt.getDimensions() and assign that to the dimension list instead.

**Review by aaime — CHANGES_REQUESTED** [review]
Looks good, small issue in the code, there is also a formatting issue, see the QA build results.

File: src/extension/wps-download/src/main/java/org/geoserver/wps/gs/download/RasterEstimator.java (hunk 1 above)
Comment (sergitopo) [reply src/extension/wps-download/src/main/java/org/geoserver/wps/gs/download/RasterEstimator.java:206]: Sorry Andrea, where is the construction of a new object in these lines?

Comment (aaime) [reply src/extension/wps-download/src/main/java/org/geoserver/wps/gs/download/RasterEstimator.java:206]: Indeed there is none ("no construction of new objects needed") so no need to guard  the statement

**Comment by sergitopo** [comment]
> Looks good, small issue in the code, there is also a formatting issue, see the QA build results.
Solved in latest commit.

**Comment by aaime** [comment]
Looks good now. Found a typo, and QA build is failing on PMD:

```
2022-02-15T10:25:54.7695676Z 10:25:54,769 [INFO] PMD Failure: org.geoserver.wps.gs.download.RasterEstimator:11 Rule:wildcards Priority:3 No Wildcard Imports.
```

Caused by:

``
import org.geoserver.catalog.*;
``

See also:
- [the coding convention](https://google.github.io/styleguide/javaguide.html#s3.3.1-wildcard-imports)
- https://stackoverflow.com/questions/147454/why-is-using-a-wild-card-with-a-java-import-statement-bad

[You can setup IntelliJ not to create wildcard imports](https://www.jetbrains.com/help/idea/creating-and-optimizing-imports.html#disable-wildcard-imports).

**Pull request merged**